FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,322 @@
+                        ResourceCount = resourceCount,
+                    };
+
+                    jobCount += 1;
+                }
+
+                _logger.LogInformation($"Splitting {resourceType} jobs, finish split {jobCount} jobs. use : {(DateTimeOffset.Now - splittingStartTime).TotalMilliseconds} milliseconds.");
+            }
+        }
+
+        private async Task<SortedDictionary<DateTimeOffset, int>> InitializeAnchorListAsync(string resourceType, DateTimeOffset? startTime, DateTimeOffset endTime, int totalCount, CancellationToken cancellationToken)
+        {
+            var anchorList = new SortedDictionary<DateTimeOffset, int>();
+            if (startTime != null)
+            {
+                anchorList[(DateTimeOffset)startTime] = 0;
+            }
+
+            // Set isDescending parameter false to get first timestmp.
+            var nextTimestamp = await GetNextTimestamp(resourceType, startTime, endTime, false, cancellationToken);
+
+            // Set isDescending parameter true to get last timestmp.
+            var lastTimestamp = await GetNextTimestamp(resourceType, startTime, endTime, true, cancellationToken);
+
+            if (nextTimestamp != null)
+            {
+                anchorList[(DateTimeOffset)nextTimestamp] = 0;
+            }
+
+            if (lastTimestamp != null)
+            {
+                // the value of anchor is resource counts less than the time stamp.
+                // Add 1 milliseond on the last resource time stamp.
+                anchorList[((DateTimeOffset)lastTimestamp).AddMilliseconds(1)] = totalCount;
+            }
+            else
+            {
+                anchorList[endTime] = totalCount;
+            }
+
+            return anchorList;
+        }
+
+        // get the lastUpdated time stamp of next resource.
+        private async Task<DateTimeOffset?> GetNextTimestamp(string resourceType, DateTimeOffset? start, DateTimeOffset end, bool isDescending, CancellationToken cancellationToken)
+        {
+            List<KeyValuePair<string, string>> parameters = new ()
+            {
+                new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"lt{end.ToInstantString()}"),
+                new KeyValuePair<string, string>(FhirApiConstants.PageCountKey, FhirApiPageCount.Single.ToString("d")),
+                new KeyValuePair<string, string>(FhirApiConstants.TypeKey, resourceType),
+                isDescending
+                ? new KeyValuePair<string, string>(FhirApiConstants.SortKey, LastUpdatedApiParameterDesc)
+                : new KeyValuePair<string, string>(FhirApiConstants.SortKey, LastUpdatedApiParameter),
+            };
+
+            if (start != null)
+            {
+                parameters.Add(new KeyValuePair<string, string>(FhirApiConstants.LastUpdatedKey, $"ge{((DateTimeOffset)start).ToInstantString()}"));
+            }
+
+            var searchOptions = new BaseSearchOptions(null, parameters);
```

Comment:
maybe wrap these operations in a function that takes parameters as input and JObject as output.

Reply:
Maybe we can refine it later.